src/test/java/com/eighthlight/SocketWrapperSpy.java

```diff
@@ -0,0 +1,52 @@
+package com.eighthlight;
+
+import java.io.IOException;
+import java.io.*;
+
+public class SocketWrapperSpy implements SocketInterface {
+    private boolean createdAndListened = false;
+    private boolean stopped = false;
+    private BufferedReader in;
+    private String receivedData;
+    private String sentData;
+
+
+    public SocketWrapperSpy(BufferedReader in){
+        this.in = in;
+    }
+
+    public SocketWrapperSpy(){
+        this(new BufferedReader(new StringReader("")));
+    }
+
+    public void createAndListen(int port) throws IOException {
+        createdAndListened = true;
+    }
+
+    public String receiveData() throws IOException {
```

Comment:
For any method that `throws`, your spy should be configurable so that it throws an exception when that method is called. For example:

```java
class SocketWrapperSpy : SocketInterface {
  public IOException createAndListenException;
  private bool createAndListenCalled = false;

  public void createAndListen(int port) throws IOException {
    createAndListenCalled = true;
    if (createAndListenException != null) {
      throw createAndListenException
    }
  }
}
```